composer/trainer/trainer.py

```diff
@@ -962,7 +965,8 @@ def _train_loop(self) -> None:
                     self.engine.run_event(Event.EPOCH_START)
                     self.logger.data_epoch({"epoch": int(self.state.timer.epoch)})
 
-                if isinstance(self.state.train_dataloader.sampler, torch.utils.data.DistributedSampler):
+                if hasattr(self.state.train_dataloader, "sampler") and isinstance(self.state.train_dataloader.sampler,
```

Comment:
Same as above: Can you add a `# TODO` in the comments, saying that the hasattr check will be removed in #424?

Reply:
will do.